corehq/apps/integration/payments/utils.py

```diff
@@ -0,0 +1,33 @@
+from datetime import datetime
+
+from corehq.apps.users.models import WebUser
+from corehq.apps.hqcase.api.updates import handle_case_update
+
+
+def verify_payment_cases(domain, case_ids: list, verifying_user: WebUser) -> tuple[int, int]:
+    payment_properties_update = {
+        'momo_payment_verified': str(True),
+        'momo_payment_verified_on_utc': str(datetime.now()),
+        'momo_payment_verified_by': verifying_user.username,
+        'momo_payment_verified_by_user_id': verifying_user.user_id,
+    }
+    cases_updates = _get_cases_updates(case_ids, payment_properties_update)
+
+    _, updated_cases = handle_case_update(
+        domain, cases_updates, verifying_user, 'momo_payment_verified', is_creation=False,
+    )
+    success_count = len(updated_cases)
```

Comment:
`handle_case_update()` doesn't chunk case updates. In case it's possible to submit an unreasonable number of cases, I'd break `case_ids` into chunks. Maybe something like:

```suggestion
    success_count = 0
    for case_id_chunk in chunked(case_ids, CHUNK_SIZE):
        cases_updates = _get_cases_updates(case_id_chunk, payment_properties_update)
        _, updated_cases = handle_case_update(
            domain,
            cases_updates,
            verifying_user,
            'momo_payment_verified',
            is_creation=False,
        )
        success_count += len(updated_cases)
```

Reply:
The report only allows up to 100 cases to be submitted at once, but now that I'm thinking about it I should not rely on an arbitrary feature on the UI for chunking. I'll add the necessary logic to chunk the cases.

Reply:
https://github.com/dimagi/commcare-hq/pull/35915/commits/58d5dd5a10a68071fd144990d4a2a68b9202b62f